pgjdbc/src/test/java/org/postgresql/test/jdbc2/DatabaseMetaDataTest.java

```diff
@@ -246,7 +249,9 @@ public void testCrossReference() throws Exception {
 
       int keySeq = rs.getInt("KEY_SEQ");
       assertEquals(j, keySeq);
+      numRows += 1;
     }
+    assertEquals(2, numRows);
```

Comment:
Technically speaking, this should have been `assertEquals("dbmd.getCrossReference  should return 2 rows since primary key m_pkey was created as (m,n)", 2, numRows)`

It does not hurt much, but it does bother when it fails like "expected: 2, got: 3".

The same for `assertEquals` above.

Something like `assertEquals("FK column " + j, expectedFkColumnNames[j - 1], fkColumnName);` would make failure messages much more convenient.

Thanks for the understanding.

Reply:
Good point!